elastalert/elastalert.py

```diff
@@ -776,7 +776,9 @@ def alert(self, matches, rule, alert_time=None):
         # Run the alerts
         alert_sent = False
         alert_exception = None
+        alert_pipeline = {}
         for alert in rule['alert']:
+            alert.pipeline = alert_pipeline
```

Comment:
alert_pipeline appears to always be an empty dictionary. Either turn this into a proper constant (ie, just name it ALERT_PIPELINE, to signify its place as a constant), or eliminate it entirely and set alert.pipeline to {} on every run.

An extra caveat here is alert_pipeline is a pointer to the same dictionary. This may cause some surprising behavior if the caller does not use a deep copy: you'll end up with the exact same pointer propagated to every rule you cut in your loop! I suspect that is not what you desire here.

As such, I vote for eliminating the extra variable to remove this surprising behavior, or if you are relying upon it, adding a comment explicitly calling out the pointer reuse.